Extend PodResources API for Dynamic Resource Allocation (#39978)
kubernetes/website: content/en/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md

    repeated string device_ids = 2;
    TopologyInfo topology = 3;
}

// DynamicResource contains information about the devices assigned to a container by Dynamic Resource Allocation
message DynamicResource {
    string class_name = 1;
    string claim_name = 2;
    string claim_namespace = 3;
    repeated ClaimResource claim_resources = 4;
}

> // ClaimResource contains per plugin resource information

Reviewer: ```suggestion
// ClaimResource contains per-plugin resource information
```

Author: done

[feature gate](/docs/reference/command-line-tools-reference/feature-gates/) to be enabled.
It is enabled by default starting with Kubernetes 1.15 and is v1 since Kubernetes 1.20.

### `Get` gRPC endpoint {#grpc-endpoint-get}

{{< feature-state state="alpa" for_k8s_version="v1.27" >}}

The `Get` endpoint provides information on resources of a running pod. It exposes similar information as
describe in the `List` endpoint. The `Get` endpoint requires `pod_name` and `pod_namesapce` of the running

> pod.

Reviewer: By the way, the design to use `_` in the field name is a bad decision against the convention to use camelCase names.

> pod.

Author: so this pod_namespace and pod_name taken from the proto file syntax which is underscore. For golang it will be camelCase according to golang conventions. I updated in the commit.

> pod.

Author: > @moshe010 Would you mind responding to @tengqm when you get the chance? Tomorrow, April 4th, is the **Docs complete — All PRs reviewed and ready to merge** deadline.

sorry for the late response I was OOO on Monday.

```gRPC
// GetPodResourcesRequest contains information about the pod
message GetPodResourcesRequest {
    string pod_name = 1;
    string pod_namespace = 2;
}
```

> To enable this feature `kubelet` must be started with the following flag:

Reviewer: ```suggestion
To enable this feature, you must start your kubelet services with the following flag:
```

```
--feature-gates=KubeletPodResourcesGet=true
```

> The `Get` enpoint can provide information on dynamic resources
> of a running pod allocated by the dynamic resource allocation API. To enable this feature `kubelet`

Reviewer: ```suggestion
The `Get` endpoint can provide Pod information related to dynamic resources
allocated by the dynamic resource allocation API. To enable this feature, you must
```

Author: done

> must be started with the following flags:

Reviewer: ```suggestion
ensure your kubelet services are started with the following flags:
```

Author: done
